app/controllers/menus_controller.rb

```diff
@@ -0,0 +1,36 @@
+class MenusController < ApplicationController
+  before_action :build_menu!, only: %i[new create]
+  before_action :find_menu!, only: %i[show edit update destroy]
+
+  def index
+    @menus = Menu.all
+  end
+
+  def show
+    @menu = MenuFacade.new(@menu)
+  end
+
+  def create
+    @menu.save
+    redirect_to menu_path(@menu), notice: 'Menu created!'
+  end
+
+  def update
+    redirect_to menu_path(@menu), notice: 'Menu updated!'
+  end
+
+  def destroy
+    @menu.destroy
+    redirect_to menus_path, notice: 'Menu destroyed!'
+  end
+
+  private
+
+  def build_menu!
+    @menu = Menu.new
```

Comment:
I don't like the way this controller is implemented.
Consider using Service / Operation-like types of objects and avoid using `before_actions` that much